keras_cv/layers/preprocessing/mix_up.py

```diff
@@ -49,15 +50,26 @@ def _sample_from_beta(alpha, beta, shape):
     def _batch_augment(self, inputs):
         images = inputs.get("images", None)
         labels = inputs.get("labels", None)
-        if images is None or labels is None:
+        bounding_boxes = inputs.get("bounding_boxes", None)
+        if images is None or (labels is None and bounding_boxes is None):
             raise ValueError(
                 "MixUp expects inputs in a dictionary with format "
-                '{"images": images, "labels": labels}.'
+                '{"images": images, "labels": labels}. or'
+                '{"images": images, "bounding_boxes": bounding_boxes}'
                 f"Got: inputs = {inputs}"
             )
-        images, labels = self._update_labels(*self._mixup(images, labels))
+        images, lambda_sample, permutation_order = self._mixup(images)
+        if labels is not None:
+            images, labels = self._update_labels(
+                images, labels, lambda_sample, permutation_order
+            )
+            inputs["labels"] = labels
+        if bounding_boxes is not None:
```

Comment:
You'll also need to consider the case where bounding_boxes AND labels are provided.  This is a supported use case.

Reply:
It is covered. In case both are not `None`, both the `if` bodies will be traversed.

In fact, I have tested using this case as well. Have a look at the changes in the `mix_up_test. py` file. I hope that is what you mean.

Reply:
I think we can also remove the `images` argument now from _update_labels and _update_bounding boxes!

Reply:
But yes looks like it works!  I was just confused by the fact that it looks like _update_labels and _update_bounding_boxes updates images (though it doesn't actually!)

Reply:
True, there was no use of `images` in  `_update_labels` and `_update_bounding_boxes`. Removed it now!

Reply:
Thanks!